salt/edx/edx_mysql.sls

```diff
@@ -1,20 +1,29 @@
 #!jinja|yaml
-{% from "shared/edx/mitx.jinja" import edx with context %}
-{% set purposes = salt.grains.get('purpose') %}
+
+{% from "orchestrate/aws_env_macro.jinja" import ENVIRONMENT with context %}
+{% set environment = salt.pillar.get('environments:{}'.format(ENVIRONMENT)) %}
+{% set purposes = environment.purposes %}
 {% set edxlocal_databases = {
   'XQUEUE_MYSQL_DB_NAME': 'xqueue',
   'EDXAPP_MYSQL_DB_NAME': 'edxapp',
   }
 %}
-{% set mysql_user = salt.pillar.get('edx:ansible_vars:EDXAPP_MYSQL_USER') %}
-{% set mysql_password = salt.pillar.get('edx:ansible_vars:EDXAPP_MYSQL_PASSWORD') %}
+{% set edxapp_mysql_creds = salt.vault.read(
+    'mysql-{env}/creds/edxapp-{purpose}'.format(
+        env=environment,
+        purpose=purpose)) %}
```

Comment:
This needs to be inside the for loop because the purpose variable hasn't been set at this point.